server/src/com/thoughtworks/go/config/update/AgentsEntityConfigUpdateCommand.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2016 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.config.update;
+
+import com.thoughtworks.go.config.*;
+import com.thoughtworks.go.config.commands.EntityConfigUpdateCommand;
+import com.thoughtworks.go.config.exceptions.NoSuchAgentException;
+import com.thoughtworks.go.config.exceptions.NoSuchEnvironmentException;
+import com.thoughtworks.go.server.domain.Username;
+import com.thoughtworks.go.server.service.GoConfigService;
+import com.thoughtworks.go.server.service.result.HttpOperationResult;
+import com.thoughtworks.go.serverhealth.HealthStateScope;
+import com.thoughtworks.go.serverhealth.HealthStateType;
+import com.thoughtworks.go.util.TriState;
+import com.thoughtworks.go.validation.AgentConfigsUpdateValidator;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class AgentsEntityConfigUpdateCommand implements EntityConfigUpdateCommand<Agents> {
+    private final Username username;
+    private final HttpOperationResult result;
+    private final List<String> uuids;
+    private final List<String> environmentsToAdd;
+    private final List<String> environmentsToRemove;
+    private final TriState enable;
+    private final List<String> resourcesToAdd;
+    private final List<String> resourcesToRemove;
+    private GoConfigService goConfigService;
+    public Agents agents;
+
+    public AgentsEntityConfigUpdateCommand(Username username, HttpOperationResult result, List<String> uuids, List<String> environmentsToAdd, List<String> environmentsToRemove, TriState enable, List<String> resourcesToAdd, List<String> resourcesToRemove, GoConfigService goConfigService) {
+        this.username = username;
+        this.result = result;
+        this.uuids = uuids;
+        this.environmentsToAdd = environmentsToAdd;
+        this.environmentsToRemove = environmentsToRemove;
+        this.enable = enable;
+        this.resourcesToAdd = resourcesToAdd;
+        this.resourcesToRemove = resourcesToRemove;
+        this.goConfigService = goConfigService;
+    }
+
+    @Override
+    public boolean canContinue(CruiseConfig cruiseConfig) {
+        if (goConfigService.isAdministrator(username.getUsername().toString())) {
```

Comment:
Create an overloaded method ```isAdministrator(CaseInseusername.getUsername)``` and refactor the existing method to call that for string based parameters (other methods in GoConfigService follow this pattern)